content/en/docs/tasks/inject-data-application/define-environment-variable-container.md

```diff
@@ -32,6 +32,12 @@ The `env` and `envFrom` fields have different effects.
  are set as environment variables for the container.
  You can also specify a common prefix string.
 
+{{< note >}}
+The allowed character range of env name is within C_IDENTIFIER,
+after enable the `RelaxedEnvironmentVariableValidation` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/), the name of env allows a wider range of characters,
```

Comment:
I would recommend removing line 36 and rewording this note.

```suggestion
If the `RelaxedEnvironmentVariableValidation` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled, all [printable ASCII characters](https://www.ascii-code.com/characters/printable-characters) except "=" may be used for environment variable names.
```

Reply:
Though is that accurate? There seems to be other printable characters that would be problematic as environment variables.

Reply:
Your description is very correct, we have done enough tests on it.